### PR TITLE
Flake8

### PR DESCRIPTION
**Pull request by benvand**
## 29.0.0

PR: [#338](https://github.com/alphagov/digitalmarketplace-utils/pull/338)

### What changed

* Code checker is changed from PEP8 to Flake8
* Makefile commands now use hyphen ('-') not underscore ('_') to delimit words

```
test_pep8 > test-flake8
show_environment > show-environment
test_python > test-python
```
* New file Flake8 config file `.flake8`
* New dependencies in `requirements-dev.txt`
* Flake8 changes implemented (including unused imports)

Actual changes:
[flake8-report.txt](https://github.com/alphagov/digitalmarketplace-utils/files/1333075/flake8-report.txt)

### CODE REVIEW

File: tests/test_s3.py [hunk 1]
```diff
@@ -255,18 +254,18 @@ def test_list_files_order_by_last_modified(self, bucket_with_multiple_files):
...
+        path,
+        expected_path,
+        expected_ct,
+        expected_filename,
+        expected_ext,
+        timestamp,
+        expected_timestamp,
+        download_filename,
+        expected_cd,
+    ):
```
Comment (samuelhwilliams) [inline tests/test_s3.py]: Would be nice if this was just on fewer lines, if you're touching it.

File: tests/test_asset_fingerprint.py [hunk 2]
```diff
@@ -91,5 +91,5 @@ def test_hash_gets_cached(self, get_file_content_mock):
 
 class TestAssetFingerprintWithUnicode(object):
     def test_can_read_self(self):
-        string_with_unicode_character = 'Ralph’s apostrophe'
+        """This string muct contain Ralph’s apostrophe. We then try to load this file with the asset fingerprinter."""
```
Comment (samuelhwilliams) [inline tests/test_asset_fingerprint.py]: `muct` typo -> `must`

Comment (benvand) [reply tests/test_asset_fingerprint.py]: Done 👍

File: tests/test_s3.py (hunk 1 above)
Comment (benvand) [reply tests/test_s3.py]: I got told off for that last time! 😂 

https://github.com/alphagov/digitalmarketplace-api/pull/601#discussion_r125472282

Thanks for the review 🙌   I'll do it 😉

Comment (benvand) [reply tests/test_s3.py]: Doing it as per the top SO answer here: https://stackoverflow.com/a/24988191/3967444

Comment (benvand) [reply tests/test_s3.py]: Done

Comment (samuelhwilliams) [reply tests/test_s3.py]: Wish I hadn't said anything 🙈 I prefer option 1 from the SO answer!

Either way though, it's better, so all good.

**Pull request merged**